pkg/cmd/adm/restart_test.go

```diff
@@ -1,216 +1,206 @@
+			tf.ClientConfigVal = cmdtesting.DefaultClientConfig()
+
+			info, _ := runtime.SerializerInfoForMediaType(ns.SupportedMediaTypes(), runtime.ContentTypeJSON)
+			encoder := ns.EncoderForVersion(info.Serializer, rolloutGroupVersionEncoder)
+			tf.Client = &RolloutRestartRESTClient{
+				RESTClient: &fake.RESTClient{
+					GroupVersion:         rolloutGroupVersionEncoder,
+					NegotiatedSerializer: ns,
+					Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+						responseDeployment := &appsv1.Deployment{}
+						responseDeployment.Name = deployment1.Name
+						responseDeployment.Labels = make(map[string]string)
+						responseDeployment.Labels[tc.labelKey] = tc.labelValue
+						body := io.NopCloser(bytes.NewReader([]byte(runtime.EncodeOrDie(encoder, responseDeployment))))
+						return &http.Response{StatusCode: http.StatusOK, Header: cmdtesting.DefaultHeader(), Body: body}, nil
+					}),
+				},
+			}
+			tf.FakeDynamicClient.WatchReactionChain = nil
+			tf.FakeDynamicClient.AddWatchReactor("*", func(action cgtesting.Action) (handled bool, ret watch.Interface, err error) {
+				fw := watch.NewFake()
+				dep := &appsv1.Deployment{}
+				dep.Name = deployment1.Name
+				dep.Status = appsv1.DeploymentStatus{
+					Replicas:            1,
+					UpdatedReplicas:     1,
+					ReadyReplicas:       1,
+					AvailableReplicas:   1,
+					UnavailableReplicas: 0,
+					Conditions: []appsv1.DeploymentCondition{{
+						Type: appsv1.DeploymentAvailable,
+					}},
+				}
+				dep.Labels = make(map[string]string)
+				dep.Labels[tc.labelKey] = tc.labelValue
+				c, err := runtime.DefaultUnstructuredConverter.ToUnstructured(dep.DeepCopyObject())
+				if err != nil {
+					t.Errorf("unexpected err %s", err)
+				}
+				u := &unstructured.Unstructured{}
+				u.SetUnstructuredContent(c)
+				go fw.Add(u)
+				return true, fw, nil
+			})
+
+			streams, _, buf, _ := genericclioptions.NewTestIOStreams()
 			term := NewFakeTerminalWithResponse("Y")
+			pod := newPod(test.NamespacedName("toolchain-host-operator", "host-operator-controller-manager"))
+			newClient, fakeClient := NewFakeClients(t, deployment1, pod)
 			ctx := clicontext.NewCommandContext(term, newClient)
 
-			// when
-			err := restart(ctx, clusterName, "wrong-deployment")
+			//when
+			err := restartDeployment(ctx, fakeClient, namespacedName.Namespace, tf, streams)
+			if tc.labelValue == "kubesaw-controller-manager" {
+				require.NoError(t, err, "non-OLM based deployment not found in")
+				err2 := deleteAndWaitForPods(ctx, fakeClient, *deployment1, tf, streams)
+				require.NoError(t, err2)
+			} else if tc.labelValue == "codeready-toolchain" {
+				require.NoError(t, err, "OLM based deployment not found in")
+				err := restartNonOlmDeployments(*deployment1, tf, streams)
+				require.NoError(t, err)
+				//checking the output from kubectl
+				require.Contains(t, buf.String(), tc.expectedOutput)
+			}
+			err1 := checkRolloutStatus(tf, streams, tc.labelSelector)
```

Comment:
I'm not sure I understand why we call all these functions, should we just invoke `restartDeployment` and do some assertions afterwards ? 

Why do we call separately also `deleteAndWaitForPods` , `restartNonOlmDeployments` ?

Reply:
yea i have updated it now